src/libstore/derivations.cc

```diff
@@ -103,31 +103,72 @@ StorePath writeDerivation(Store & store,
 }
 
 
-/* Read string `s' from stream `str'. */
-static void expect(std::istream & str, std::string_view s)
+/* Derivation parser local to this file.  Keep in unnabed namespace
+   to unblock aggressive inlining. */
```

Comment:
```suggestion
/* Derivation parser local to this file.  Keep in unnamed namespace
   to unblock aggressive inlining. */
```

What does unblock mean? Prevent aggressive inlining? BTW pretty surprising that namespaces affect inlining...

Reply:
Oh, good point! I assumed C++ methods have interposition restrictions (effectively forbid inline) similar to C exported functions, but from superficial test it does not.

I'll drop `namespace{}` and re-test that it has no performance impact.

Reply:
keeping the anonymous namespace around the struct would be a good idea, otherwise `DerivationParserContext` could in theory conflict with like-named declarations in other files. this isn't all that likely, but it seems like good practice to prevent such things.

Reply:
I dropped the `namespace{}` to be more in line with the rest of `nix`. After measuring the performance I don't see a slowdown (yay!).